search/mongo_search.go

```diff
@@ -370,25 +371,29 @@ func (m *MongoSearcher) createChainedSearchPipelineStages(searchParam SearchPara
 	if isOr {
 		lookupRef, ok = searchParam.(*OrParam).Items[0].(*ReferenceParam)
 		if !ok {
-			panic(createInternalServerError("", ""))
+			panic(createInternalServerError("", "Chained query OR has no references"))
 		}
 	} else {
 		lookupRef = searchParam.(*ReferenceParam)
 	}
 
 	chainedRef, ok := lookupRef.Reference.(ChainedQueryReference)
 	if !ok {
-		panic(createInternalServerError("", ""))
+		panic(createInternalServerError("", "ReferenceParam is not of type ChainedQueryReference"))
 	}
 
+	// We need a $lookup stage for each path, followed by one $match stage
+	stages := make([]bson.M, len(lookupRef.getInfo().Paths)+1)
 	collectionName := models.PluralizeLowerResourceName(chainedRef.Type)
 
-	stages[0] = bson.M{"$lookup": bson.M{
-		"from":         collectionName,
-		"localField":   convertSearchPathToMongoField(lookupRef.getInfo().Paths[0].Path) + ".referenceid",
-		"foreignField": "_id",
-		"as":           "_" + collectionName,
-	}}
+	for i, path := range lookupRef.Paths {
+		stages[i] = bson.M{"$lookup": bson.M{
+			"from":         collectionName,
+			"localField":   convertSearchPathToMongoField(path.Path) + ".referenceid",
+			"foreignField": "_id",
+			"as":           "_lookup" + strconv.Itoa(i),
```

Comment:
OK.  In case you find yourself needing to do something like this again, I'd also consider `fmt.Sprintf("_lookup%d", i)`.